model/src/main/java/org/cloudfoundry/identity/uaa/client/Jackson2ArrayOrStringDeserializer.java

```diff
@@ -0,0 +1,40 @@
+package org.cloudfoundry.identity.uaa.client;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.type.SimpleType;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+@SuppressWarnings("serial")
+public class Jackson2ArrayOrStringDeserializer extends StdDeserializer<Set<String>> {
```

Comment:
I can see that this class and the above one, and probably many other classes in this PR are re-implementation of what is in `spring-security-oauth2`. However, I had to look into `spring-security-oauth2` to figure that out. I think that documenting about that in the source code a little bit will be helpful.

Reply:
OK, after going through all the files in this PR, I realize that this PR is indeed all about replacing the classes in `spring-security-oauth2`. So a little bit of Javadoc in the new class sources will be definitely help people who read the codes later. That is all.

Reply:
This PR was independent so far from rest.
I tried to do similar things with Exceptions and OAuth2RestTemplate but UAA and spring-security-oauth2 are so deep connected that it does not work. This means the next https://github.com/cloudfoundry/uaa/pull/2813 will be then the final one - of course without dependency to spring-security-oauth2. But therefore I moved all classes into on namespace with UAA so that it gets visible what classes we need

Reply:
Cool. It appears that documentations were added to the classes. Thanks.